orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,203 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
+	}
+
+	outputByLines := strings.Split(strings.TrimRight(versionOutputStr, "\n"), "\n")
+	if len(outputByLines) < 1 {
+		return "", errors.New("expected number of lines is not present")
+	}
+
+	rawVersionStr := strings.TrimSpace(strings.ToLower(outputByLines[0]))
+	if !strings.HasPrefix(rawVersionStr, expectedPrefix) {
+		return "", errors.New("expected version prefix is not present")
+	}
+
+	// getting the actual version string
+	versionStr := strings.ReplaceAll(rawVersionStr, expectedPrefix, "")
+	if len(versionStr) == 0 {
+		return "", errors.New("expected version information is not present")
+	}
+
+	return versionStr, nil
+}
+
+// fixSymlinkNotPresent fixes the issue where the symlink to the orbit service binary is not present
+// this is a workaround for the issue described here https://github.com/fleetdm/fleet/issues/10300
+func fixSymlinkNotPresent() error {
+	// getting current working directory
+	execPath, err := getWorkingPath()
+	if err != nil {
+		return err
+	}
+
+	// getting the path to orbit service binary
+	orbitPath := execPath + "\\..\\bin\\orbit\\orbit.exe"
+
+	// gathering target orbit version
+	versionOrbit, err := getOrbitVersion(orbitPath)
+	if err != nil {
+		return fmt.Errorf("getting orbit version: %w", err)
+	}
+
+	// checking if target orbit has the problematic logic
+	if versionOrbit != "1.6.0" && versionOrbit != "1.7.0" {
+		return nil
+	}
+
+	// checking if the orbit service binary symlink needs to be regenerated
+	_, err = os.Readlink(orbitPath)
+
+	// if there are no errors, it means that the symlink is present, nothing to do
+	if err == nil {
+		return nil
+	}
+
+	// handling error by renaming the locked binary file, marking it for deletion on reboot and
+	// regenerating the symlink
+	if !errors.Is(err, os.ErrNotExist) {
+
+		// symlink is going to be recreated - so better check that symlink target is present
+		targetSymlinkPath := ""
+		stableOrbitPath := execPath + "\\..\\bin\\orbit\\windows\\stable\\orbit.exe"
+		edgeOrbitPath := execPath + "\\..\\bin\\orbit\\windows\\edge\\orbit.exe"
+
+		if _, err := os.Stat(stableOrbitPath); err == nil {
+			targetSymlinkPath = stableOrbitPath
+		} else if _, err := os.Stat(edgeOrbitPath); err == nil {
+			targetSymlinkPath = edgeOrbitPath
+		}
+
+		if len(targetSymlinkPath) == 0 {
+			return errors.New("symlink target cannot be found")
+		}
+
+		// We are now about to perform a sensitive operation
+		// by first renaming current orbit file and then recreating recreating the symlink
+
+		// renaming locked binary to a different file, the process will keep running, but it will be renamed
+		// target process is not terminated on purpose to avoid potential erros
+		targetOrbitPath := orbitPath + "." + strings.ToUpper(uuid.New().String())
+		if err := os.Rename(orbitPath, targetOrbitPath); err != nil {
+			return fmt.Errorf("rename: %w", err)
+		}
+
+		// regenerating the symlink
+		if err := os.Symlink(targetSymlinkPath, orbitPath); err != nil {
```

Comment:
This is similar to the auto update code, which brings me to the question that maybe this only happens on fresh installations of the `msi`? (and the link not being preserved)

Reply:
we need to investigate this. I noticed the symlink creation after a clean install, so I'm still not sure what's happening

Reply:
> I noticed the symlink creation after a clean install

Where is this?

Reply:
I noticed the behavior through testing. I'm going to check the code to better understand this

Reply:
How do I know if a file is a symlink in Windows, `C:\Program Files\Orbit\bin\orbit\orbit.exe` on my VM (after a fresh installation) doesn't seem to be a symbolic link / reparse point:
```
C:\Program Files\Orbit\bin\orbit>dir /a
 Volume in drive C has no label.
 Volume Serial Number is 904A-65D7

 Directory of C:\Program Files\Orbit\bin\orbit

03/07/2023  03:21 PM    <DIR>          .
03/07/2023  03:21 PM    <DIR>          ..
03/07/2023  06:19 PM        33,640,448 orbit.exe
03/07/2023  03:21 PM    <DIR>          windows
               1 File(s)     33,640,448 bytes
               3 Dir(s)  31,487,401,984 bytes free
```

Reply:
you can use powershell for this `Get-ChildItem 'C:\Program Files\Orbit\bin\orbit\'  | Select-Object -Property *`

Reply:
```
PS C:\Users\Lucas Rodriguez> Get-ChildItem 'C:\Program Files\Orbit\bin\orbit\orbit.exe' | Select-Object -Property *

PSPath            : Microsoft.PowerShell.Core\FileSystem::C:\Program Files\Orbit\bin\orbit\orbit.exe
PSParentPath      : Microsoft.PowerShell.Core\FileSystem::C:\Program Files\Orbit\bin\orbit
PSChildName       : orbit.exe
PSDrive           : C
PSProvider        : Microsoft.PowerShell.Core\FileSystem
PSIsContainer     : False
Mode              : -a----
VersionInfo       : File:             C:\Program Files\Orbit\bin\orbit\orbit.exe
                    InternalName:
                    OriginalFilename:
                    FileVersion:
                    FileDescription:
                    Product:
                    ProductVersion:
                    Debug:            False
                    Patched:          False
                    PreRelease:       False
                    PrivateBuild:     False
                    SpecialBuild:     False
                    Language:

BaseName          : orbit
Target            : {}
LinkType          :
Name              : orbit.exe
Length            : 33640448
DirectoryName     : C:\Program Files\Orbit\bin\orbit
Directory         : C:\Program Files\Orbit\bin\orbit
IsReadOnly        : False
Exists            : True
FullName          : C:\Program Files\Orbit\bin\orbit\orbit.exe
Extension         : .exe
CreationTime      : 3/7/2023 6:19:24 PM
CreationTimeUtc   : 3/7/2023 9:19:24 PM
LastAccessTime    : 3/7/2023 8:25:29 PM
LastAccessTimeUtc : 3/7/2023 11:25:29 PM
LastWriteTime     : 3/7/2023 6:19:24 PM
LastWriteTimeUtc  : 3/7/2023 9:19:24 PM
Attributes        : Archive
```

Doesn't look like a link, right?

Reply:
Ok, I now understand what happened. 

So you are correct that this bug only happens on fresh installations of Orbit 1.6.0 or 1.7.0. The bug will not trigger if previous Orbit versions were used for fresh installs. This last thing happened on my env, I was using a snapshot with an MSI that contained an old orbit version. Symlink on this scenario gets generated and the regular orbit file dropped by the installer gets deleted. Self-update worked as expected here, and orbit gets updated to the latest. 

The issue will not happen if a pre-1.6.0 version of Orbit is installed on a machine where the self-update ran at least once. In this case, the symlink will be already there, and Orbit can update to 1.6.0 and 1.7.0 without any issues.

The issue only happens if Orbit 1.6.0 or Orbit 1.7.0 is installed through MSI

I've put more details of this [here](https://github.com/fleetdm/fleet/issues/10300#issuecomment-1459262529)